hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieMergedLogRecordReader.java

```diff
@@ -230,6 +242,34 @@ public void processNextRecord(T record, Map<String, Object> metadata) throws IOE
     }
   }
 
+  @Override
+  public void processNextRecordWithPosition(T record,
+                                            Map<String, Object> metadata,
+                                            int sequenceNo,
+                                            List<Integer> positions) throws IOException {
```

Comment:
I think the record positions are `long` right?

Reply:
Do we need long? 2 Billion records for a base file is enough?